src/app/shared/object-list/selectable-list/selectable-list.actions.ts

```diff
@@ -0,0 +1,84 @@
+import { Action } from '@ngrx/store';
+import { type } from '../../ngrx/type';
+import { ListableObject } from '../../object-collection/shared/listable-object.model';
+
+/**
+ * For each action type in an action group, make a simple
+ * enum object for all of this group's action types.
+ *
+ * The 'type' utility function coerces strings into string
+ * literal types and runs a simple check to guarantee all
+ * action types in the application are unique.
+ */
+export const SelectableListActionTypes = {
+  SELECT: type('dspace/selectable-lists/SELECT'),
+  SELECT_SINGLE: type('dspace/selectable-lists/SELECT_SINGLE'),
+  DESELECT: type('dspace/selectable-lists/DESELECT'),
+  DESELECT_SINGLE: type('dspace/selectable-lists/DESELECT_SINGLE'),
+  SET_SELECTION: type('dspace/selectable-lists/SET_SELECTION'),
+  DESELECT_ALL: type('dspace/selectable-lists/DESELECT_ALL')
+};
+
+/* tslint:disable:max-classes-per-file */
+export abstract class SelectableListAction implements Action {
+  // tslint:disable-next-line:no-shadowed-variable
+  constructor(public type, public id: string) {
+  }
+}
+
+/**
+ * Used to select an item in a the selectable list
+ */
+export class SelectableListSelectAction extends SelectableListAction {
```

Comment:
Could you similarly add comments for all the Action classes in this file?  See the ones below